### PR TITLE
Remove resolvers to repositories that don't exist

### LINKED ISSUE #2195: Classpaths refers to repositories that don't actually exist
Classpaths (Defaults.scala) refers to repositories that don't exist:
- [typesafeSnapshots](https://github.com/sbt/sbt/blob/v0.13.9/main/src/main/scala/sbt/Defaults.scala#L1651) (Ivy)
- [sbtPluginSnapshots](https://github.com/sbt/sbt/blob/v0.13.9/main/src/main/scala/sbt/Defaults.scala#L1659)

There's also [typesafeRepo(status: String)](https://github.com/sbt/sbt/blob/v0.13.9/ivy/src/main/scala/sbt/Resolver.scala#L207) (Maven) which alludes to a possible Typesafe Maven snapshots repository.

The URLs are:
- https://repo.typesafe.com/typesafe/snapshots/
- https://repo.typesafe.com/typesafe/ivy-snapshots/
- https://repo.scala-sbt.org/scalasbt/sbt-plugin-snapshots/

Which redirect to:
- https://dl.bintray.com/typesafe/snapshots/
- https://dl.bintray.com/typesafe/ivy-snapshots/
- https://dl.bintray.com/sbt/sbt-plugin-snapshots/

which then 404.

Note, Bintray doesn't support snapshots.

### PR DESCRIPTION
**Pull request by dwijnand**
Fixes #2195

### CODE REVIEW

File: main/src/main/scala/sbt/Defaults.scala [hunk 1]
```diff
@@ -123,7 +123,7 @@ object Defaults extends BuildCommon {
       retrieveManagedSync :== false,
       configurationsToRetrieve :== None,
       scalaOrganization :== ScalaArtifacts.Organization,
-      sbtResolver := { if (sbtVersion.value endsWith "-SNAPSHOT") Classpaths.typesafeSnapshots else Classpaths.typesafeReleases },
+      sbtResolver := Classpaths.typesafeReleases,
```
Comment (eed3si9n) [inline main/src/main/scala/sbt/Defaults.scala]: sbt is now published to Maven Central for the releases, we might keep using https://bintray.com/sbt/ivy-snapshots for the nightly.

Comment (dwijnand) [inline main/src/main/scala/sbt/Defaults.scala]: `Classpaths.typesafeSnapshots` is https://repo.typesafe.com/typesafe/ivy-snapshots/ which is https://dl.bintray.com/typesafe/ivy-snapshots/.

https://bintray.com/sbt/ivy-snapshots (which is https://repo.scala-sbt.org/scalasbt/ivy-snapshots/) isn't modeled anywhere currently. Do you want me to add it?

Comment (eed3si9n) [inline main/src/main/scala/sbt/Defaults.scala]: Yea. Let's do it.

**Comment by eed3si9n** [comment]
So this is still pending right?

**Comment by dwijnand** [comment]
Yep, I'll be pushing an -Xn release of sbt/library-management to finish this one off.

**Pull request merged**